Shared/Headers/FocusEventListener.hpp

```diff
@@ -8,17 +8,44 @@
 #include <string>
 #include <functional>
 #include <thread>
+#include <nanomsg/nn.h>
+#include <nanomsg/pubsub.h>
+#include "FocusEvent.pb.h"
 
+template<class TPayload>
 class FocusEventListener {
 private:
-	std::function<void(std::string clientId, std::string payload)> _onMessage;
-	std::unique_ptr<std::thread> _eventListenerThread;
-	int _socketSUB;
-	static void RunReceive(int socketSUB, std::function<void(std::string clientId, std::string payload)> *onMessage, bool shouldUnpack = false);
+    std::function<void(std::string clientId, TPayload &payload)> _onMessage;
+    std::unique_ptr<std::thread> _eventListenerThread;
+    int _socketSUB;
+
+    static void RunReceive(int socketSUB, std::function<void(std::string clientId, TPayload &)> *onMessage) {
+        while (true) {
+            char *buf = NULL;
+            int bytes = nn_recv(socketSUB, &buf, NN_MSG, 0);
+            std::string recv = std::string(buf);
```

Comment:
You can use `std::string recv(buf);` to avoid one copy in general. (Although this isn't applicable thanks to std::string move assignement)